website/api/security-model.md

```diff
@@ -0,0 +1,108 @@
+
+Proofs generated by the Recursion Prover and proofs generated by the STARK-to-SNARK Prover reveal no information beyond the contents of the [receipt claim].
+Even with unlimited computing power, no secrets can be extracted from proofs posted on-chain.
+
+**Warning 1**: Whoever is generating the proofs can see the secret data. Users may choose to use any of these provers locally to ensure their data stays private.
+
+**Warning 2:** Proofs generated by the **RISC-V Prover** that haven't been passed through the **Recursion Prover** leak information about the length of execution.
+Passing proofs through the Recursion Prover resolves this warning: recursion proofs leak no information about execution length.
+
+# Cryptographic Security
+
+In analyzing the cryptographic security of our system, we consider two primary questions:
+
+1. Can a malicious user who doesn't know a valid [execution trace] for the given [receipt claim] create a fake proof that will trick the verifier?
+2. Can a malicious user extract secret information from a proof posted on-chain?
+
+The first question is about the **soundness** of the protocol, and the second question is whether the protocol is **zero-knowledge**.
+
+Soundness is often quantified in terms of “[bits]” — our system currently targets 98 bits of security.
+
+| Prover                | Cryptographic Assumptions                                                                 | Bits of Security | Quantum Safe? |
+| --------------------- | ----------------------------------------------------------------------------------------- | ---------------- | ------------- |
+| RISC-V Prover         | - Random Oracle Model <br/> - Toy Problem Conjecture                                      | 98               | Yes           |
+| Recursion Prover      | - Random Oracle Model <br/> - Toy Problem Conjecture                                      | 99               | Yes           |
+| STARK-to-SNARK Prover | - Security of elliptic curve pairing over BN254. <br/> - Knowledge of Exponent assumption | 100+             | No            |
+
+The Toy Problem conjecture, specified in detail in the [ethSTARK documentation], says that the best known attack on STARK proof systems is the best possible attack.
+
+## The STARK Circuits
+
+The RISC-V Prover and the Recursion Prover both use STARK-based protocols, which are not known to be vulnerable to attacks via quantum computers.
+
+### How secure are the STARK Provers?
+
+We estimate that it would cost over 400 billion USD of computational resources to construct a fake proof for one of our STARK circuits.
+
+These estimates are based on the approach used in [this article by Justin Thaler].
+The napkin math here is as follows:
+
+- Thaler estimated $1.6 million to launch a viable attack against an 80 bit STARK system.
+- 98 bits is $2^{18}$ times more secure than 80 bits.
+
+The analysis for bits of security for the RISC-V and Recursion Prover can be found in the [security calculator].
```

Comment:
The `[security calculator]` link is broken

Reply:
good catch -- should be fixed now